tools/sphere.c

```diff
@@ -0,0 +1,82 @@
+// Mystical Sphere of Slooping
+// Gon
+// 2017-04
+
+inherit NewThingCode;
+inherit AnsiTellCode;
+
+#include <ansi.h>
+
+#define SLOOPEE_SHAD "/usr/gon/tools/sloopee_shad"
+#define SLOOPER_SHAD "/usr/gon/tools/slooper_shad"
+
+// Functions
+int query_sloop();
+int do_sloop();
+int do_touch();
+int do_gaze();
+
+void extra_create() {
+    set_short( "A Glowing Sphere" );
+    set_ids( ({"a glowing sphere", "glowing sphere", "sphere", 
+        "a sphere" }) );
+    set_long( "The sphere is glowing a deep purple that "
+        "seems to pulse rhythmically, almost as if the object "
+        "was breathing. The surface appears to be smooth "
+        "and without any marks. It makes you slightly uneasy "
+        "as you look upon it, almost as if it were aware of your "
+        "presence..." );
+    set_gettable( 0 );
+}
+
+extra_init() {
+    add_action( "do_sloop", "sloop" );
+    add_action( "do_gaze", "gaze" );
+    add_action( "do_touch", "touch" );
+}
+
+// Apply Slooper and Sloopee shadows, respectively
+int do_sloop( string arg ) {
+    if ( !arg ) {
+        return notify_fail( "What do you want to sloop?" );
+    }
+    object slooper = THISP;
+    if ( !IsWizard( slooper ) ) {
+        return notify_fail( "You cannot summon the power needed." );
+    }
+    object sloopee = FINDP ( arg );
+    if ( !sloopee ) {
+        return notify_fail( "You can only sloop a player." );
+    }
+    if ( slooper == sloopee ) {
+        return notify_fail( "Your head begins to spin and you quickly look away "
+            "from the sphere. Maybe you shouldn't try slooping yourself?" );
+    }
+    if ( GetOrdLevel( slooper ) <= GetOrdLevel( sloopee ) ) {
+        ansi_tell( slooper, sprintf("You have not the power to sloop %s" 
+                ,FINDP( arg )->query_name() ), BOLD_PURPLE );
+        return 1;
+    }
+    else {
```

Comment:
If you return in every conditional up above, you don't necessarily need an `else`. You can do the meat of the logic outside of the `if` statements.